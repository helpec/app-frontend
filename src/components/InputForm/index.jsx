import React from 'react';


class InputForm extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            inputValuer: props.value,
            submitted: props.submitted
        };
        this.handleChange = this.handleChange.bind(this);
    }

    handleChange(e) {
      const { value } = e.target;
      this.setState({ inputValuer: value });
      this.props.callbeack({ [this.props.name]: value })
    }
    
    render() {
        const { inputValuer } = this.state;
        const { name, label, submitted } = this.props;
        return (
          <div className={'control-group' + (submitted && !inputValuer ? ' has-error' : '')}>
              <label className="control-label" htmlFor={name}> {label} </label>
              <div className="controls">
                  <input type="text" className="form-control" name={name} 
                         value={inputValuer} onChange={this.handleChange} />
                  {submitted && !inputValuer &&
                      <div className="help-block">O campo {label} Ã© obrigatorio</div>
                  }
              </div>
          </div>
        );
    }
}
export default InputForm