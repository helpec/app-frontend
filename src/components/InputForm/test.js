import React from 'react';
import renderer from 'react-test-renderer';

import InputForm from '.';

test('Testar componente de InputForm da aplicação', () => {

    const component = renderer.create(
      <InputForm />
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

});
