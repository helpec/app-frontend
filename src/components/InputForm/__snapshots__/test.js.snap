// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testar componente de InputForm da aplicação 1`] = `
<div
  className="control-group"
>
  <label
    className="control-label"
  >
     
     
  </label>
  <div
    className="controls"
  >
    <input
      className="form-control"
      onChange={[Function]}
      type="text"
    />
  </div>
</div>
`;
